model/src/DA/Finance/Types.daml

```diff
@@ -63,3 +63,19 @@ data MasterAgreement = MasterAgreement
     party2 : Party
       -- ^ Allows to specify choices of the second counterparty.
   deriving (Eq, Show)
+
+-- | A record for templates which can be locked. Also includes a set of
+--  parties which may unlock a locked template.
+data Lock = Lock
+  with
+    by : Party
+      -- ^ The party which performed the locking action.
+    at : Time
+      -- ^ The time at which the locking action was executed.
+    reason : Text
+      -- ^ The purpose of locking.
+    relatingId : Id
```

Comment:
I wonder if `Id` is the right type here. We tend to overuse it and it somewhat ties the locking mechanism to use cases that use the `Id` type. I'd argue using a `Text` here is more general purpose usable.

Reply:
That's a valid point 👍